front/src/applications/stdcm/components/StdcmForm/StdcmOperationalPoint.tsx

```diff
@@ -73,65 +99,40 @@ const StdcmOperationalPoint = ({ location, pathStepId, disabled }: StdcmOperatio
             return acc;
           },
           [] as { label: string; id: string }[]
-        ),
-    [location, sortedSearchResults]
-  );
-
-  const dispatchNewPoint = (p?: SearchResultItemOperationalPoint) => {
-    if (p && location && p.ch === location.secondary_code && p.uic === location.uic) return;
-    const newLocation = p
-      ? {
-          name: p.name,
-          secondary_code: p.ch,
-          uic: p.uic,
-          coordinates: p.geographic.coordinates as [number, number],
-        }
-      : undefined;
-    dispatch(updateStdcmPathStep({ id: pathStepId, updates: { location: newLocation } }));
-  };
-
-  const updateSelectedPoint = (
-    refList: SearchResultItemOperationalPoint[],
-    selectedUic: number,
-    selectedChCode?: string
-  ) => {
-    const newPoint = refList.find(
-      (pr) => pr.uic === selectedUic && (selectedChCode ? pr.ch === selectedChCode : true)
-    );
-    dispatchNewPoint(newPoint);
-  };
-
-  const onSelectSuggestion = (selectedSuggestion?: Option) => {
-    if (!selectedSuggestion) {
-      setSearchTerm('');
-      return;
+        );
+      setChSuggestions(newChSuggestions);
+    } else {
+      setChSuggestions([]);
     }
-    const { value: suggestionName, uic } = selectedSuggestion;
-    setSearchTerm(suggestionName);
-    updateSelectedPoint(sortedSearchResults, uic);
+    dispatch(updateStdcmPathStep({ id: pathStepId, updates: { location: selectedSuggestion } }));
   };
 
-  const onSelectChCodeFilter = (selectedChCode?: { id: string }) => {
-    setChCodeFilter(selectedChCode?.id);
-    if (location) {
-      updateSelectedPoint(sortedSearchResults, location.uic, selectedChCode?.id);
+  const onSelectCh = (selectedChCode?: { id: string }) => {
+    if (location && selectedChCode) {
+      dispatch(
+        updateStdcmPathStep({
+          id: pathStepId,
+          updates: { location: { ...location, secondary_code: selectedChCode?.id } },
+        })
+      );
     }
   };
 
-  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+  const onCiInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
the more react idiomatic ways is to name props `onSomething` and the function called `handleSomething`

here why not just `handleCiChange`